src/runtime_src/core/pcie/driver/linux/include/mailbox_proto.h

```diff
@@ -338,22 +338,22 @@ struct xcl_mailbox_bitstream_kaddr {
  * @target_freqs: array of target clock frequencies (max clocks: 16)
  */
 struct xcl_mailbox_clock_freqscaling {
-	unsigned int region;
-	unsigned short target_freqs[16];
+	uint32_t region;
+	uint16_t target_freqs[16];
 };
 
 #define XCL_MB_REQ_FLAG_RESPONSE	(1 << 0)
 #define XCL_MB_REQ_FLAG_REQUEST		(1 << 1)
 /**
  * struct mailbox_req - mailbox request message header
- * @req: opcode
+ * @req: opcode (enum xcl_mailbox_request)
  * @flags: flags of this message
  * @data: payload of variable length
  */
 struct xcl_mailbox_req {
 	uint64_t flags;
-	enum xcl_mailbox_request req;
-	char data[1]; /* variable length of payload */
+	int32_t req;
+	int8_t data[4]; /* variable length of payload, explicitly padded to int32_t */
```

Comment:
If I was doing this, I would have just added the padding as a separate definition - same comment for `struct xcl_sw_chan`.

Reply:
Hi @richardtobin-xlnx ,

This is what I initially thought about. But if I keep "data" as the last member, it is more consistent with the fact that it is the starting point of a variable length chunk of payload in the same data structure buffer. Adding another member after it confuses me. Anyway, I changed the type (vs making it an array) but still keep it as the last one.

I also fixed the other data structures that you pointed out. Thanks for pointing me to the compiler tool. It was very helpful. Now I can see that all data structure size are the same without or without packing on both 32 and 64 bit platforms.

Please take another look.